debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/ObjectToSparseDataHashTable.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright IBM Corp. and others 2025
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ */
+package com.ibm.j9ddr.vm29.j9;
+
+import com.ibm.j9ddr.CorruptDataException;
+import com.ibm.j9ddr.vm29.pointer.generated.J9HashTablePointer;
+import com.ibm.j9ddr.vm29.pointer.generated.MM_SparseDataTableEntryPointer;
+import com.ibm.j9ddr.vm29.pointer.VoidPointer;
+import com.ibm.j9ddr.vm29.types.UDATA;
+
+/**
+ * ObjectToSparseDataHashTable is used for accessing hash tables {@link HashTable}
+ * in SparseDataTableEntry {@link MM_SparseDataTableEntryPointer} (e.g. iterator)。
+ */
+public class ObjectToSparseDataHashTable extends HashTable_V1<MM_SparseDataTableEntryPointer>
```

Comment:
This class should be `final` and should have an explicit (`private`) constructor.